freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaPostInstallService.java

```diff
@@ -65,11 +65,14 @@
     @Inject
     private FreeIpaService freeIpaService;
 
-    public void postInstallFreeIpa(Long stackId) throws Exception {
-        LOGGER.debug("Performing post-install configuration for stack {}", stackId);
+    public void postInstallFreeIpa(Long stackId, boolean fullPostInstall) throws Exception {
+        LOGGER.debug("Performing post-install configuration for stack {}. {}.", stackId, fullPostInstall ? "Full post install" : "Partial post install");
         Stack stack = stackService.getStackById(stackId);
         FreeIpaClient freeIpaClient = freeIpaClientFactory.getFreeIpaClientForStack(stack);
         freeIpaTopologyService.updateReplicationTopology(stackId, freeIpaClient);
+        if (!fullPostInstall) {
+            return;
+        }
```

Comment:
could you refactor this? like move the rest of the method to a separate one, and put that call in the if.